### ISSUE TITLE
can't get lock on C_START_FISHING_MINIGAME and C_END_FISHING_MINIGAME for protocol 359882

### ISSUE THREAD

**Issue by DaXmAn** [SoliaRdi/auto-fishing#285]
when I enable logger for client only and manually catch fish, I see this stack:

[13:42:48:822] ->          	 C_USE_ITEM    (id 42729)
{
    "gameId": "XXX",
    "id": XXX,
    "dbid": "0",
    "target": "0",
    "amount": 1,
    "dest": {
        "x": 0,
        "y": 0,
        "z": 0
    },
    "loc": {
        "x": XXX,
        "y": XXX,
        "z": XXX
    },
    "w": 0,
    "unk1": 0,
    "unk2": 0,
    "unk3": 0,
    "unk4": true
}
[removed hex for obvious reasons]

[13:42:48:867] ->          	    (id 27484) null
00000004: 00000000 02000000 00000000                                               ............

10005c6b000000000200000000000000
[13:42:50:862] ->          	    (id 43591) null
00000004: 07000000 8F030000 BC820110 72270300 5FAD940E 00000000 03000000           ........<...r'.._-..........

200047aa070000008f030000bc820110722703005fad940e0000000003000000
[13:43:24:662] ->          	    (id 40739) null
00000004: 06000000 1D010000                                                        ........

0c00239f060000001d010000
[13:43:32:426] ->          	    (id 22579) null
00000004: 06000000 0A000000 01                                                     .........

0d003358060000000a00000001

When I was tailing the log it seems that when I  click F I got 40739 and when I finished minigame I got 22579, but when I use these opcodes autofishing do not start :( just waits like regular autofishing

**Comment by SoliaRdi** [comment]
defs was changed

**Comment by DaXmAn** [comment]
do you have new defs by any chance?

**Comment by Foks5000** [comment]
First word is a packet counter (independent of each other)

**Comment by SoliaRdi** [comment]
idk yet what this dynamic value that smh related to counter works. and how to calc it

**Comment by SoliaRdi** [comment]
Seems like you can just hardcode new fields ,but im not sure that this solution is safe. So gonna wait more

**Comment by maryannew** [comment]
make new protocol
protocol.359882.map

C_END_FISHING_MINIGAME = 22579
C_PUT_WARE_ITEM = 34823
C_RQ_ADD_ITEM_TO_DECOMPOSITION_CONTRACT = 62893
C_START_FISHING_MINIGAME = 40739
C_STORE_COMMIT = 43678
C_STORE_SELL_ADD_BASKET = 34507
S_FISHING_BITE = 24212
S_FISHING_CATCH = 42961
S_RP_ADD_ITEM_TO_DECOMPOSITION_CONTRACT = 23731
S_START_FISHING_MINIGAME = 29037
S_STORE_BASKET = 50056

**Comment by zokiopp** [comment]
any ETA on re enabling the mod?

**Comment by SoliaRdi** [comment]
I can release today. But idk if my always same packet counter is safe xD

**Comment by DaXmAn** [comment]
Go for it ;) but add some kind of popup warning in toolbox proxy saying that this may not be safe? :) once module will load & also message in toolbox console

**Comment by SoliaRdi** [comment]
update pushed. New defs was added

**Comment by Nykkyta** [comment]
what do you mean by 'safe/unsafe'? What's the risk of using the mod as it is now?
thank you for all the good hard work, really appreciated

**Comment by SoliaRdi** [comment]
Both questions - idk. Only time will tell. Previous versions was fully client emulation without any packet wise exploits, now client packets and mod are not same

**Comment by DaXmAn** [comment]
so the only uknown is in these 2 packets? The counter paramter? which is random when game client sends it? Or is it in order?

mod.send('C_START_FISHING_MINIGAME', mod.majorPatchVersion>=88?2:1, {
	counter:1,
	unk:15
});

mod.send('C_END_FISHING_MINIGAME', mod.majorPatchVersion>=88?2:1, {
	counter:1,
	unk:24,
	success: true
});

**Comment by SoliaRdi** [comment]
unk relative to counter smh(so same unk for same counter). And counter getting reset only after client restart

**Comment by SoliaRdi** [comment]
maybe i should make unk random(didnt test if this works), and leave counter as client send. Idk

**Comment by DaXmAn** [comment]
wait a sec, can't you just capture what client want to send and just use that? Somehow trigger these events on client side faster or something?

**Comment by SoliaRdi** [comment]
?_?

**Comment by DaXmAn** [comment]
Just thinking without complete knowledge about possibilities here. I mean proxy is a man in a middle, between game client and server. So this may not be possible. Is there a way to trigger something in the client from proxy? So the client will do real operation and return proper packet?

**Comment by SoliaRdi** [comment]
This is possible. But this is another story and i wont do this.

**Comment by DaXmAn** [comment]
If you like, I can write C# app and publish code on git, so anybody could download binaries or compile themselfs. It would be for in .NET framework. If not, let's forget about it and wait what will happen next with current solution :)

**Comment by SoliaRdi** [comment]
not needed at all. Nobody will use this

**Comment by DaXmAn** [comment]
Alright 👍

**Comment by Lambda11** [comment]
> unk relative to counter smh(so same unk for same counter). And counter getting reset only after client restart

does it (counter or unk) even change during game session or we could just ask user to first catch one fish himself and take correct data about counter and unk from hooked `C_START_FISHING_MINIGAME` , `C_END_FISHING_MINIGAME` ?

**Comment by SoliaRdi** [comment]
counter and unk are repeated per client run(only full restart reset them).

**Comment by Owyn** [comment]
so both values are always the same whole game session?

 - and unk aren't always 15 & 24 ? also random as two counters for them (which stays the same whole game)?

doesn't sound hard to properly emulate then

**Comment by SoliaRdi** [comment]
unk always changing with counter. I didnt test if we can send just random unk for any counter

**Comment by Owyn** [comment]
Why even send random stuff when you can send correct stuff same as game itself sends?

**Comment by SoliaRdi** [comment]
how you send correct C_START_MINIGAME? unk are diff in all packets with same counter

**Comment by Owyn** [comment]
> counter and unk are repeated per client run(only full restart reset them).

but you said here they are always the same unless you restart the game, no?

**Comment by SoliaRdi** [comment]
They diff in all packets. Same within one opcode

**Comment by Owyn** [comment]
Then the plan to send correct ones is:
hook needed opcodes
make user trigger them
use their data from now on in this session (from one opcode to the same opcode)
- should all be correct, no?

**Comment by SoliaRdi** [comment]
its impossible. Thats why i hardcode just first manual fishing values
